fdbserver/workloads/Cycle.actor.cpp

```diff
@@ -220,10 +236,41 @@ struct CycleWorkload : TestWorkload {
 			// One client checks the validity of the cycle
 			state Transaction tr(cx);
 			state int retryCount = 0;
+			state Version minVersion =
+			    wait(ensureNoCommitsInFlight(cx, self)); // For comparing the txnStateStore and the storage servers
 			loop {
 				try {
 					state Version v = wait( tr.getReadVersion() );
-					Standalone<RangeResultRef> data = wait(tr.getRange(firstGreaterOrEqual(doubleToTestKey(0.0, self->keyPrefix)), firstGreaterOrEqual(doubleToTestKey(1.0, self->keyPrefix)), self->nodeCount + 1));
+					state Standalone<RangeResultRef> data = wait(
+					    tr.getRange(firstGreaterOrEqual(doubleToTestKey(0.0, self->keyPrefix)),
+					                firstGreaterOrEqual(doubleToTestKey(1.0, self->keyPrefix)), self->nodeCount + 1));
+					if (self->keyPrefix.startsWith("\xff/TESTONLYtxnStateStore/"_sr)) {
+						loop {
+							state DebugReadTxnStateStoreReply rep =
+							    wait(debugReadTxnStateStore(cx,
+							                                KeyRangeRef(doubleToTestKey(0.0, self->keyPrefix),
+							                                            doubleToTestKey(1.0, self->keyPrefix)),
+							                                self->nodeCount + 1));
+							if (rep.version < minVersion) {
+								wait(delay(1));
+								continue;
+							}
+							if (data != rep.kvs) {
+								for (const auto& [k, v] : data) {
```

Comment:
nit: 
Can we print out the keys from data and rep.kvs like this:
index i: data[i], rep.kvs[I]? 

It can be easier to identify which key becomes different when failure happens.